In [127]:
import pandas as pd 
import matplotlib.pyplot as plt  
df = pd.read_csv("C:/Users/cyrin/logements-sociaux-paris/data/raw/logements-sociaux-finances-a-paris.csv", sep=';')

In [128]:
df.head()

,id_livraison,adresse_programme,code_postal,ville,annee,bs,nb_logmt_total,nb_plai,nb_plus,nb_pluscd,nb_pls,mode_real,commentaires,arrdt,nature_programme,coord_x_l93,coord_y_l93,geo_shape,geo_point_2d
0,2000011,16-20 RUE DES MEUNIERS,75012,Paris,2001,RES.URB.,62,0,62,0,0,acquisition conventionnement,NaN,12,logement familial,655823.1529,6.859495e+06,"{""coordinates"": [2.398175442412243, 48.8339981...","48.83399817959342, 2.398175442412243"
1,2000035,25 RUE DES ANNELETS,75019,Paris,2001,RIVP,18,10,8,0,0,acquisition réhabilitation,NaN,19,logement familial,655225.8330,6.864433e+06,"{""coordinates"": [2.3895183752327043, 48.878361...","48.8783619998301, 2.3895183752327043"
2,2000044,LOT D1B - 86 RUE DES HAIES,75020,Paris,2001,RIVP,6,0,6,0,0,acquisition réhabilitation,NaN,20,logement familial,656258.4385,6.861828e+06,"{""coordinates"": [2.403865260493333, 48.8550036...","48.85500366971155, 2.403865260493333"
3,2000048,31 QUAI DE VALMY,75010,Paris,2001,PARIS HABITAT,28,0,22,0,6,acquisition conventionnement,NaN,10,logement familial,653551.4260,6.863384e+06,"{""coordinates"": [2.366804047083707, 48.8688074...","48.86880747974437, 2.366804047083707"
4,2000062,19 RUE DES PLANTES,75014,Paris,2001,PARIS HABITAT,38,0,15,0,23,acquisition conventionnement,NaN,14,logement familial,650356.4410,6.859151e+06,"{""coordinates"": [2.3237456731730535, 48.830499...","48.83049983027532, 2.3237456731730535"


In [129]:
print("Nombre de lignes :", df.shape[0])
print("Nombre de colonnes :", df.shape[1])

Nombre de lignes : 4174
Nombre de colonnes : 19


In [130]:
df.columns.to_list()

['id_livraison',
 'adresse_programme',
 'code_postal',
 'ville',
 'annee',
 'bs',
 'nb_logmt_total',
 'nb_plai',
 'nb_plus',
 'nb_pluscd',
 'nb_pls',
 'mode_real',
 'commentaires',
 'arrdt',
 'nature_programme',
 'coord_x_l93',
 'coord_y_l93',
 'geo_shape',
 'geo_point_2d']

In [131]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 4174 entries, 0 to 4173
Data columns (total 19 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id_livraison       4174 non-null   str    
 1   adresse_programme  4174 non-null   str    
 2   code_postal        4174 non-null   str    
 3   ville              4174 non-null   str    
 4   annee              4174 non-null   int64  
 5   bs                 4174 non-null   str    
 6   nb_logmt_total     4174 non-null   int64  
 7   nb_plai            4174 non-null   int64  
 8   nb_plus            4174 non-null   int64  
 9   nb_pluscd          4174 non-null   int64  
 10  nb_pls             4174 non-null   int64  
 11  mode_real          4174 non-null   str    
 12  commentaires       0 non-null      float64
 13  arrdt              4174 non-null   int64  
 14  nature_programme   4174 non-null   str    
 15  coord_x_l93        4174 non-null   float64
 16  coord_y_l93        4174 non-null   

In [132]:
df.dtypes

id_livraison             str
adresse_programme        str
code_postal              str
ville                    str
annee                  int64
bs                       str
nb_logmt_total         int64
nb_plai                int64
nb_plus                int64
nb_pluscd              int64
nb_pls                 int64
mode_real                str
commentaires         float64
arrdt                  int64
nature_programme         str
coord_x_l93          float64
coord_y_l93          float64
geo_shape                str
geo_point_2d             str
dtype: object

In [133]:
print(f"lignes dupliquées :{df.duplicated().sum()}")

lignes dupliquées :0


In [134]:
print("\nValeurs manquantes par colonne :")
missing = df.isnull().sum()
missing_pct = (missing / len(df) * 100).round(2)
print(pd.DataFrame({"missing_count": missing, "missing_pct": missing_pct}))


Valeurs manquantes par colonne :
                   missing_count  missing_pct
id_livraison                   0          0.0
adresse_programme              0          0.0
code_postal                    0          0.0
ville                          0          0.0
annee                          0          0.0
bs                             0          0.0
nb_logmt_total                 0          0.0
nb_plai                        0          0.0
nb_plus                        0          0.0
nb_pluscd                      0          0.0
nb_pls                         0          0.0
mode_real                      0          0.0
commentaires                4174        100.0
arrdt                          0          0.0
nature_programme               0          0.0
coord_x_l93                    0          0.0
coord_y_l93                    0          0.0
geo_shape                      0          0.0
geo_point_2d                   0          0.0


In [135]:
print("\nRésumé des variables catégorielles :")
print(df.describe(include=["object", "category"]))


Résumé des variables catégorielles :
       id_livraison adresse_programme code_postal  ville            bs  \
count          4174              4174        4174   4174          4174   
unique         4141              3705          21      1            66   
top         2002559    36, RUE SIBUET       75018  Paris  ELOGIE SIEMP   
freq              3                 7         520   4174          1098   

                 mode_real   nature_programme  \
count                 4174               4174   
unique                   3                 13   
top     construction neuve  logement familial   
freq                  1638               3551   

                                                geo_shape  \
count                                                4174   
unique                                               3492   
top     {"coordinates": [2.40247300337046, 48.84379734...   
freq                                                   12   

                                geo_poi

C:\Users\cyrin\AppData\Local\Temp\ipykernel_28084\1956021498.py:2: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  print(df.describe(include=["object", "category"]))


In [136]:


pd.set_option('display.max_rows', None)

print(df['bs'].value_counts(dropna=False))

bs
ELOGIE SIEMP                         1098
PARIS HABITAT                        1021
RIVP                                  783
AXIMO                                 147
IMM 3F                                123
(vide)                                109
SABLIERE                               88
SNL-PROLOG                             78
BATIGERE                               61
HSF                                    59
TOIT ET JOIE                           57
1001 VIES HABITAT                      50
EFIDIS                                 50
ADOMA                                  50
SEQENS HABITAT                         43
FREHA                                  41
HAB.CONF.                              34
RATP HABITAT                           29
3F RES                                 27
RES.L.F.                               25
HENEO                                  22
FONCIERE                               21
COALLIA                                17
ANTIN                          

In [137]:
print("\nNombre de valeurs uniques par colonne :")
print(df.nunique())


Nombre de valeurs uniques par colonne :
id_livraison         4141
adresse_programme    3705
code_postal            21
ville                   1
annee                  24
bs                     66
nb_logmt_total        243
nb_plai               143
nb_plus               167
nb_pluscd              24
nb_pls                147
mode_real               3
commentaires            0
arrdt                  20
nature_programme       13
coord_x_l93          3490
coord_y_l93          3491
geo_shape            3492
geo_point_2d         3492
dtype: int64


In [138]:
pd.set_option('display.max_rows', None)

print(df['mode_real'].value_counts(dropna=False))

mode_real
construction neuve              1638
acquisition réhabilitation      1594
acquisition conventionnement     942
Name: count, dtype: int64


In [139]:
pd.set_option('display.max_rows', None)

print(df['nature_programme'].value_counts(dropna=False))

nature_programme
logement familial                                                                    3551
résidence étudiante                                                                   173
foyer pour jeunes travailleurs ou résidence pour jeunes actifs                        114
maison relais                                                                          84
résidence sociale pour démunis                                                         81
foyer de travailleurs migrants                                                         36
résidence sociale issue du plan de traitement des foyers de travailleurs migrants      33
foyer pour personnes handicapées                                                       32
établissement d'hébergement pour personnes âgées dépendantes                           26
centre d'hébergement d'urgence                                                         26
résidence pour personnes âgées                                                     

In [140]:
pd.set_option('display.max_rows', None)

print(df['annee'].value_counts(dropna=False))

annee
2010    279
2009    269
2008    268
2007    267
2011    206
2013    205
2012    201
2017    199
2015    196
2018    181
2014    177
2016    171
2006    161
2004    154
2005    152
2019    150
2021    144
2023    144
2022    129
2020    123
2002    107
2003    105
2024    101
2001     85
Name: count, dtype: int64


In [141]:
pd.set_option('display.max_rows', None)

print(df['arrdt'].value_counts(dropna=False))

arrdt
18    520
20    494
15    386
13    364
19    309
11    294
12    289
17    282
10    225
14    221
16    139
9     121
4      98
3      90
5      86
1      72
2      57
6      52
8      46
7      29
Name: count, dtype: int64


In [142]:
import numpy as np
numeric_cols = df.select_dtypes(include=[np.number]).columns
for col in numeric_cols:
    print(f"\n{col}: min={df[col].min()}, max={df[col].max()}, mean={df[col].mean():.2f}")


annee: min=2001, max=2024, mean=2012.27

nb_logmt_total: min=-8, max=1332, mean=30.32

nb_plai: min=-8, max=400, mean=8.29

nb_plus: min=-1, max=799, mean=12.51

nb_pluscd: min=0, max=92, mean=0.17

nb_pls: min=0, max=533, mean=9.34

commentaires: min=nan, max=nan, mean=nan

arrdt: min=1, max=20, mean=13.88

coord_x_l93: min=645313.119, max=657018.241, mean=652446.84

coord_y_l93: min=6857636.1173, max=6867035.1139, mean=6862553.57


In [143]:
print("\nMatrice de corrélation :")
print(df[numeric_cols].corr())


Matrice de corrélation :
                   annee  nb_logmt_total   nb_plai   nb_plus  nb_pluscd  \
annee           1.000000        0.014854  0.048402 -0.030948  -0.039503   
nb_logmt_total  0.014854        1.000000  0.574810  0.801997   0.024814   
nb_plai         0.048402        0.574810  1.000000  0.229557  -0.012430   
nb_plus        -0.030948        0.801997  0.229557  1.000000  -0.020783   
nb_pluscd      -0.039503        0.024814 -0.012430 -0.020783   1.000000   
nb_pls          0.029346        0.711376  0.109751  0.361354   0.000708   
commentaires         NaN             NaN       NaN       NaN        NaN   
arrdt           0.001919        0.067869  0.024863  0.012929   0.020390   
coord_x_l93    -0.020527        0.010338 -0.016097 -0.023908  -0.011441   
coord_y_l93    -0.070529       -0.063309 -0.025253 -0.063928   0.050132   

                  nb_pls  commentaires     arrdt  coord_x_l93  coord_y_l93  
annee           0.029346           NaN  0.001919    -0.020527    -0.070

In [144]:
print("\nUtilisation mémoire :")
print(df.memory_usage(deep=True))


Utilisation mémoire :
Index                   132
id_livraison          62401
adresse_programme    145149
code_postal           54267
ville                 54262
annee                 33392
bs                    72774
nb_logmt_total        33392
nb_plai               33392
nb_plus               33392
nb_pluscd             33392
nb_pls                33392
mode_real            132290
commentaires          33392
arrdt                 33392
nature_programme     116303
coord_x_l93           33392
coord_y_l93           33392
geo_shape            337024
geo_point_2d         186760
dtype: int64


In [145]:
print("STATISTIQUES DESCRIPTIVES (colonnes numériques)")
df.describe()

STATISTIQUES DESCRIPTIVES (colonnes numériques)


,annee,nb_logmt_total,nb_plai,nb_plus,nb_pluscd,nb_pls,commentaires,arrdt,coord_x_l93,coord_y_l93
count,4174.000000,4174.000000,4174.000000,4174.000000,4174.000000,4174.000000,0.0,4174.000000,4174.000000,4.174000e+03
mean,2012.272640,30.317202,8.293244,12.512698,0.174892,9.336368,NaN,13.882846,652446.840649,6.862554e+06
std,6.107193,62.877886,25.114113,33.987187,2.557760,29.690293,NaN,4.955583,2668.942840,2.469549e+03
min,2001.000000,-8.000000,-8.000000,-1.000000,0.000000,0.000000,NaN,1.000000,645313.119000,6.857636e+06
25%,2008.000000,4.000000,0.000000,0.000000,0.000000,0.000000,NaN,11.000000,650667.110000,6.860459e+06
50%,2012.000000,13.000000,1.000000,3.000000,0.000000,0.000000,NaN,15.000000,652626.340500,6.862616e+06
75%,2017.000000,30.000000,6.000000,11.000000,0.000000,6.000000,NaN,18.000000,654546.157900,6.864759e+06
max,2024.000000,1332.000000,400.000000,799.000000,92.000000,533.000000,NaN,20.000000,657018.241000,6.867035e+06


In [146]:
neg_cols = ['nb_logmt_total', 'nb_plai', 'nb_plus', 'nb_pluscd', 'nb_pls']
neg_mask = (df[neg_cols] < 0).any(axis=1)
print(f"{neg_mask.sum()} lignes avec au moins un nombre négatif\n")
neg_rows = df.loc[neg_mask, ['id_livraison', 'adresse_programme', 'annee', 'bs',
                              'mode_real', 'nb_logmt_total', 'nb_plai', 'nb_plus',
                              'nb_pluscd', 'nb_pls']]

print(neg_rows)



2 lignes avec au moins un nombre négatif

     id_livraison                              adresse_programme  annee  \
2528      2000587  56-58, RUE LEON FROT 3, RUE CARRIERE MAINGUET   2010   
3457      2000900       16-22, RUE FERNAND LEGER (FOYER MURIERS)   2010   

                bs                   mode_real  nb_logmt_total  nb_plai  \
2528  ELOGIE SIEMP          construction neuve               5        6   
3457       COALLIA  acquisition réhabilitation              -8       -8   

      nb_plus  nb_pluscd  nb_pls  
2528       -1          0       0  
3457        0          0       0  


In [147]:
print("\nPar mode_real:")
print(neg_rows["mode_real"].value_counts())
print("\n par bs:")
print(neg_rows["bs"].value_counts())
print("\nPar annee:")
print(neg_rows["annee"].value_counts().sort_index())


Par mode_real:
mode_real
construction neuve            1
acquisition réhabilitation    1
Name: count, dtype: int64

 par bs:
bs
ELOGIE SIEMP    1
COALLIA         1
Name: count, dtype: int64

Par annee:
annee
2010    2
Name: count, dtype: int64


In [148]:
check = neg_rows.copy()
check['sum_categories'] = check[['nb_plai', 'nb_plus', 'nb_pluscd','nb_pls']].sum(axis=1)
check['matches_total'] = check['sum_categories'] == check['nb_logmt_total']
print("\nVérification de cohérence (total == somme des sous-catégories) :")
print(check[['id_livraison', 'nb_logmt_total', 'sum_categories', 'matches_total']])


Vérification de cohérence (total == somme des sous-catégories) :
     id_livraison  nb_logmt_total  sum_categories  matches_total
2528      2000587               5               5           True
3457      2000900              -8              -8           True


In [149]:
dupe_ids = df[df.duplicated('id_livraison', keep=False)].sort_values('id_livraison')
print(f"{dupe_ids['id_livraison'].nunique()} identifiants distincts apparaissent plus d'une fois "
      f"({len(dupe_ids)} lignes au total)\n")

print(dupe_ids[['id_livraison', 'adresse_programme', 'annee', 'mode_real',
                 'nb_logmt_total']].to_string())



32 identifiants distincts apparaissent plus d'une fois (65 lignes au total)

     id_livraison                                                     adresse_programme  annee                     mode_real  nb_logmt_total
1656      2000403                                                        25, RUE MORAND   2004    acquisition réhabilitation              29
1963      2000403                                                        25, RUE MORAND   2004            construction neuve               9
1972      2000587                         56-58, RUE LEON FROT 3, RUE CARRIERE MAINGUET   2005    acquisition réhabilitation              15
2528      2000587                         56-58, RUE LEON FROT 3, RUE CARRIERE MAINGUET   2010            construction neuve               5
1560      2000735                                       161-161BIS RUE DE LA CONVENTION   2006    acquisition réhabilitation               5
2052      2000735                                       161-161BIS RUE DE LA 

In [150]:
# Indiquer si les lignes dupliquées partagent la même adresse et la même année (probablement un vrai doublon)
# ou si elles diffèrent (probablement une livraison distincte ou réalisée en plusieurs phases)
dupe_ids['same_addr_year'] = dupe_ids.duplicated(
    subset=['id_livraison', 'adresse_programme', 'annee'], keep=False
)
print("\nLignes où l'adresse et l'année sont identiques pour le même id_livraison "
      "(probablement de vrais doublons) :")
print(dupe_ids[dupe_ids['same_addr_year']])

print("\nLignes où l'adresse ou l'année diffère (probablement des livraisons distinctes légitimes) :")
print(dupe_ids[~dupe_ids['same_addr_year']])


Lignes où l'adresse et l'année sont identiques pour le même id_livraison (probablement de vrais doublons) :
     id_livraison                     adresse_programme code_postal  ville  \
1656      2000403                        25, RUE MORAND       75011  Paris   
1963      2000403                        25, RUE MORAND       75011  Paris   
1560      2000735       161-161BIS RUE DE LA CONVENTION       75015  Paris   
2052      2000735       161-161BIS RUE DE LA CONVENTION       75015  Paris   
1041      2000767   84, AVENUE DU DOCTEUR ARNOLD NETTER       75012  Paris   
3087      2000767   84, AVENUE DU DOCTEUR ARNOLD NETTER       75012  Paris   
1005      2003476            32, RUE DU GENERAL GUILHEM       75011  Paris   
3895      2003476            32, RUE DU GENERAL GUILHEM       75011  Paris   
511       2004147  CASERNE CHALVIDAN 45-57  BD EXELMANS       75016  Paris   
4166      2004147  CASERNE CHALVIDAN 45-57  BD EXELMANS       75016  Paris   
196       2004216                

In [151]:
## Vérification de cohérence : un même programme (id_livraison) 
## doit avoir un seul mode_real, sinon l'agrégation par mode serait faussée

incoherent = df.groupby('id_livraison')['mode_real'].nunique()
n_incoherent = (incoherent > 1).sum()
print(f"Programmes avec mode_real incohérent entre leurs lignes : {n_incoherent}")

if n_incoherent > 0:
    ids_incoherents = incoherent[incoherent > 1].index
    print(df[df['id_livraison'].isin(ids_incoherents)]
          [['id_livraison', 'adresse_programme', 'annee', 'mode_real']])

Programmes avec mode_real incohérent entre leurs lignes : 7
     id_livraison                                  adresse_programme  annee  \
196       2004216                                     18 RUE SURCOUF   2024   
409       2003725  9 QUAI SAINT BERNARD 8BIS RUE CUVIER (BARRE CA...   2024   
721       2002196                                  5-7, RUE DU SAHEL   2012   
1005      2003476                         32, RUE DU GENERAL GUILHEM   2019   
1311      2002196                                 1 RUE PAUL CRAMPEL   2024   
1656      2000403                                     25, RUE MORAND   2004   
1963      2000403                                     25, RUE MORAND   2004   
1972      2000587      56-58, RUE LEON FROT 3, RUE CARRIERE MAINGUET   2005   
2528      2000587      56-58, RUE LEON FROT 3, RUE CARRIERE MAINGUET   2010   
2645      2003725  9 QUAI SAINT BERNARD CUVIER (BARRE CASSAN JUSS...   2020   
3457      2000900           16-22, RUE FERNAND LEGER (FOYER MURIERS)   

In [152]:
## 9ter. Test plus robuste : comparer le début de l'adresse (moins sensible 
## aux petites variations de texte) plutôt que l'égalité stricte

for id_ in ids_incoherents:
    sub = df[df['id_livraison'] == id_]
    adresses = sub['adresse_programme'].unique()
    # Compare seulement les 15 premiers caractères, normalisés
    prefixes = set(a.strip().upper()[:15] for a in adresses)
    same_prefix = len(prefixes) == 1
    flag = "OK (adresse similaire)" if same_prefix else "⚠️ SUSPECT (adresses vraiment différentes)"
    print(f"id_livraison {id_} — {flag}")
    if not same_prefix:
        for a in adresses:
            print(f"   -> {a}")

id_livraison 2000403 — OK (adresse similaire)
id_livraison 2000587 — OK (adresse similaire)
id_livraison 2000900 — OK (adresse similaire)
id_livraison 2002196 — ⚠️ SUSPECT (adresses vraiment différentes)
   -> 5-7, RUE DU SAHEL
   -> 1 RUE PAUL CRAMPEL
id_livraison 2003476 — OK (adresse similaire)
id_livraison 2003725 — OK (adresse similaire)
id_livraison 2004216 — OK (adresse similaire)


In [153]:
top_programs = df.sort_values('nb_logmt_total', ascending=False).head(10)
print(top_programs[['id_livraison', 'adresse_programme', 'annee', 'bs',
                     'mode_real', 'nb_logmt_total']])

     id_livraison                                  adresse_programme  annee  \
1175      2003676                                 PORTE DE MONTREUIL   2019   
3869      2002949           92-94-98, BD PONIATOWSKI 2-4, RUE DUBOIS   2016   
3283      2002792                      1-11, RUE DE L'ABBE ROUSSELOT   2015   
2175      2002793                          1-11, SQUARE DU THIMERAIS   2015   
3894      2003460                     15 AVENUE DE LA PORTE DE VITRY   2018   
3303      2003251                       174 ET 180-188, RUE LECOURBE   2017   
1121      2002559                           128-138, BOULEVARD MURAT   2014   
3288      2002963        54-68, 70-72 ET 109-111, BOULEVARD SERURIER   2016   
2645      2003725  9 QUAI SAINT BERNARD CUVIER (BARRE CASSAN JUSS...   2020   
759       2002794                           1-8, SQUARE DU THIMERAIS   2015   

                 bs                     mode_real  nb_logmt_total  
1175  PARIS HABITAT  acquisition conventionnement            1

In [154]:
# Est-ce plausible ? Comparer à la distribution
print(f"\n99e percentile : {df['nb_logmt_total'].quantile(0.99):.0f}")
print(f"95e percentile : {df['nb_logmt_total'].quantile(0.95):.0f}")
print(f"Maximum : {df['nb_logmt_total'].max()}")

# Vérifier manuellement l'adresse (par exemple en effectuant une recherche en ligne) pour contrôler la cohérence
# l'échelle — de grands programmes de logements sociaux parisiens de plus de 1000 logements existent


99e percentile : 277
95e percentile : 115
Maximum : 1332


In [155]:
print(df['nature_programme'].value_counts())

# Tableau croisé : nature_programme influence-t-elle la distribution de mode_real ?
print("\nnature_programme x mode_real (pourcentage par ligne) :")
ct = pd.crosstab(df['nature_programme'], df['mode_real'], normalize='index')
print((ct * 100).round(1))

nature_programme
logement familial                                                                    3551
résidence étudiante                                                                   173
foyer pour jeunes travailleurs ou résidence pour jeunes actifs                        114
maison relais                                                                          84
résidence sociale pour démunis                                                         81
foyer de travailleurs migrants                                                         36
résidence sociale issue du plan de traitement des foyers de travailleurs migrants      33
foyer pour personnes handicapées                                                       32
établissement d'hébergement pour personnes âgées dépendantes                           26
centre d'hébergement d'urgence                                                         26
résidence pour personnes âgées                                                     

In [156]:
print(df.groupby('mode_real')[['nb_plai', 'nb_plus', 'nb_pluscd', 'nb_pls']]
        .agg(['sum', 'mean', 'min']))

                             nb_plai                nb_plus                 \
                                 sum       mean min     sum       mean min   
mode_real                                                                    
acquisition conventionnement    7464   7.923567   0   20582  21.849257   0   
acquisition réhabilitation      8840   5.545797  -8   10162   6.375157   0   
construction neuve             18312  11.179487   0   21484  13.115995  -1   

                             nb_pluscd               nb_pls                 
                                   sum      mean min    sum       mean min  
mode_real                                                                   
acquisition conventionnement         0  0.000000   0  18597  19.742038   0  
acquisition réhabilitation          58  0.036386   0   5200   3.262233   0  
construction neuve                 672  0.410256   0  15173   9.263126   0  


## Conclusions — Data Understanding

- **Négatifs** (`nb_plai`/`nb_plus`/`nb_logmt_total` < 0) : quelques lignes isolées, 
  probablement des corrections de déclarations antérieures. Impact négligeable 
  (2 lignes / 4174). Décision : exclues du dataset nettoyé.

- **id_livraison dupliqués** : NE SONT PAS des doublons. Ils représentent des tranches 
  de financement ou des phases de livraison distinctes d'un même programme. 
  Décision : conservées telles quelles, id_livraison n'est pas une clé unique par ligne.

- **id_livraison avec mode_real incohérent entre ses lignes** : 7 cas détectés sur le 
  dataset brut (10 lignes). Test de similarité d'adresse (préfixe normalisé) : 
  6 cas sont des tranches/phases légitimes du même programme (même adresse, à des 
  variations d'écriture près). 1 cas (id_livraison 2002196) présente deux adresses 
  réellement différentes (5-7 RUE DU SAHEL vs 1 RUE PAUL CRAMPEL, années 2012/2024), 
  suggérant une réutilisation de l'identifiant plutôt qu'un même programme. 
  Comme l'analyse se fait au niveau ligne (chaque ligne porte son propre mode_real 
  et ses comptes de financement), ceci n'affecte pas l'agrégation par mode_real. 
  Aucune action de nettoyage nécessaire ; documenté pour ne pas traiter id_livraison 
  comme identifiant de projet fiable.

- **bs = "(vide)"** : placeholder de valeur manquante (109 lignes), non utilisé dans 
  l'analyse principale. Pas d'action nécessaire.

- **Outlier nb_logmt_total = 1332** : légitime, cohérent avec les gros programmes 
  d'acquisition conventionnement (rachat de parc immobilier existant, ex. Porte de 
  Montreuil). Conservé.

- **nature_programme fortement corrélé à mode_real** (ex : résidences étudiantes 
  81% en acquisition réhabilitation, EHPAD 92% en acquisition conventionnement) : 
  facteur de confusion potentiel pour l'analyse principale. Décision : analyse 
  centrée sur `nature_programme == "logement familial"` (3551 programmes), avec 
  vérification de robustesse sur le dataset complet.